src/viam/examples/modules/simple/main.cpp

```diff
@@ -24,118 +23,126 @@
+        throw std::invalid_argument("Need socket path as command line argument");
     }
+    std::string socket_addr = argv[1];
 
     // Use set_logger_severity_from_args to set the boost trivial logger's
     // severity depending on commandline arguments.
     set_logger_severity_from_args(argc, argv);
     BOOST_LOG_TRIVIAL(debug) << "Starting module with debug level logging";
 
-    // C++ modules must handle SIGINT and SIGTERM. Make sure to create a sigset
-    // for SIGINT and SIGTERM that can be later awaited in a thread that cleanly
-    // shuts down your module. pthread_sigmask should be called near the start
-    // of main so that later threads inherit the mask.
-    sigset_t sigset;
-    sigemptyset(&sigset);
-    sigaddset(&sigset, SIGINT);
-    sigaddset(&sigset, SIGTERM);
-    pthread_sigmask(SIG_BLOCK, &sigset, NULL);
+    // C++ modules must handle SIGINT and SIGTERM. You can use the SignalManager
+    // class and its wait method to handle the correct signals.
+    SignalManager signals;
 
     API generic = Generic::static_api();
     Model m("acme", "demo", "printer");
 
     std::shared_ptr<ModelRegistration> mr = std::make_shared<ModelRegistration>(
-        ResourceType("MyModule"),
+        ResourceType("Printer"),
         generic,
         m,
-        [](Dependencies, ResourceConfig cfg) { return std::make_unique<MyModule>(cfg); },
+        [](Dependencies, ResourceConfig cfg) { return std::make_unique<Printer>(cfg); },
         // Custom validation can be done by specifying a validate function like
         // this one. Validate functions can `throw` exceptions that will be
         // returned to the parent through gRPC. Validate functions can also return
         // a vector of strings representing the implicit dependencies of the resource.
         [](ResourceConfig cfg) -> std::vector<std::string> {
-            if (cfg.attributes()->find("invalidattribute") != cfg.attributes()->end()) {
-                throw std::runtime_error(
-                    "'invalidattribute' attribute not allowed for model 'acme:demo:printer'");
-            }
-
-            return {"component1"};
+            // find_to_print will throw an error if the `to_print` attribute
+            // is missing, is not a string or is an empty string.
+            Printer::find_to_print(cfg);
+            return {};
         });
 
     Registry::register_model(mr);
 
     // The `ModuleService_` must outlive the Server, so the declaration order
     // here matters.
-    auto my_mod = std::make_shared<ModuleService_>(argv[1]);
+    auto my_mod = std::make_shared<ModuleService_>(socket_addr);
     auto server = std::make_shared<Server>();
 
     my_mod->add_model_from_registry(server, generic, m);
     my_mod->start(server);
+    BOOST_LOG_TRIVIAL(info) << "Module serving model " << m.to_string() << ", listening on "
+                            << socket_addr;
 
-    std::thread server_thread([&server, &sigset]() {
```

Comment:
I now start the server synchronously; @abe-winter's example module does this, and this simple module seems to run/shutdown just fine with the RDK. This seems to have gotten rid of the issue in https://viam.atlassian.net/browse/RSDK-4729.